mdata/aggmetric.go

```diff
@@ -342,8 +331,13 @@ func (a *AggMetric) Get(from, to uint32) (Result, error) {
 		}
 	}
 
+	if oldestChunk.First {
```

Comment:
It seems unnecessary that each chunk needs to have this `First` bool. Can you use the `t0` and `LastTs` to be able to tell whether `firstTs` falls in this range? Something like a `ContainsTimestamp` function?

Reply:
It's less about the space and more about the branching of chunk types, when from a `Chunk`s perspective it is just a `Chunk` and has no concept of `First`ness. Like, if `Chunk` was just a slice, it would be weird for the `First` member to live with the slice, right?

Reply:
funny, i also thought about this. but what I realized is that chunk.Chunk is already not a pure chunk, rather it is a "chunk as embedded in an AggMetric in metrictank", what i mean is it already has a bunch of fields for other uses within metrictank besides mere chunkness. so one more doesn't change anything IMHO.

that said at some point i want to refactor this to clarify this.

Reply:
If it doesn't bother you that's ok. But I'll point out that the other elements of the `Chunk` are actually properties relevant to a `Chunk`s internals. `First` will be the first/only property that is actually about a state outside of `Chunk`, and that's what makes it stand out as awkward to me.

Reply:
makes sense. you're looking at it from a "field describes which state" perspective.  I was looking at it "field is used for what/by what" perspective.
using the former perspective generally leads to cleaner code I think. And you're right, it looks a bit more awkward to me too now. But it's the most efficient solution so until we refactor I think it's OK.